core/src/main/java/com/vityuk/ginger/loader/PropertiesLocalizationLoader.java

```diff
@@ -85,26 +83,33 @@ private ResourcePropertyResolver load(MatchingReader reader) throws IOException
                 String value = reader.readLineUntil(LINE_SEPARATOR_MATCHER);
 
                 Matcher matcher = MAP_KEY_PATTERN.matcher(key);
+                String propertyKey;
+                String mapKey;
                 if (matcher.matches()) {
                     /*
                      This is map property of format: propertyKey[mapKey]=value
                     */
-                    String propertyKey = matcher.group(1);
-                    String mapKey = matcher.group(2);
-                    Map<String, String> propertyMap = mapProperties.get(propertyKey);
-                    if (propertyMap == null) {
-                        propertyMap = Maps.newHashMapWithExpectedSize(4);
-                        mapProperties.put(propertyKey, propertyMap);
-                    }
-                    propertyMap.put(mapKey, value);
+                    propertyKey = matcher.group(1);
+                    mapKey = matcher.group(2);
                 } else {
-                    properties.put(key, value);
+                    propertyKey = key;
+                    mapKey = "";
```

Comment:
Maybe there is a better and/or cleaner way. But with this fix there is only one "kind" of property internally. That allow to use getString(String key) without care if this is a property or a propertyMap (cf getString in interface generator)